app/services/streams/twitch_access_token/get.rb

```diff
@@ -7,11 +7,11 @@ def self.call
       end
 
       def call
-        token, exp = Rails.cache.fetch(ACCESS_TOKEN_AND_EXPIRATION_CACHE_KEY)
+        token, exp = RedisRailsCache.fetch(ACCESS_TOKEN_AND_EXPIRATION_CACHE_KEY)
```

Comment:
btw, shouldn't this be a `.read` ?

Reply:
`fetch` will attempt to read the cache, I guess technically it could be `read` for simplicity but I think it's fine to leave it.

However, and this was a miss on my part when I suggested using the expiration as the actual expiration, we should update this cache key so that it's clear it only represents the token bc now the expiration is used as the actual expiration. We also need to calculate the time until the expiration, we can't just use a Datetime for expires. Once again, that was a review miss on my part! Sorry!

In other words, here is what I believe needs to happen to iron this out with the new approach. 

```ruby 
token = RedisRailsCache.read(ACCESS_TOKEN_CACHE_KEY)

return token unless token.nil?
token, exp = get_new_token
time_til_exp = exp - Time.now # this should give you the time until expiration in seconds which is what we need
RedisRailsCache.write(ACCESS_TOKEN_CACHE_KEY, token, expires_in: time_til_exp) # This actually returns "OK" which is why we need an additional line to return the token.
token
```